rust/operator-binary/src/airflow_controller.rs

```diff
@@ -803,74 +799,17 @@ fn add_authentication_volumes_and_volume_mounts(
     cb: &mut ContainerBuilder,
     pb: &mut PodBuilder,
 ) -> Result<()> {
-    let authentication_class_name = authentication_class.metadata.name.as_ref().unwrap();
-
     match &authentication_class.spec.provider {
         AuthenticationClassProvider::Ldap(ldap) => {
-            if let Some(bind_credentials) = &ldap.bind_credentials {
-                let volume_name = format!("{authentication_class_name}-bind-credentials");
-
-                pb.add_volume(build_secret_operator_volume(
-                    &volume_name,
-                    &bind_credentials.secret_class,
-                    bind_credentials.scope.as_ref(),
-                ));
-                cb.add_volume_mount(&volume_name, format!("{SECRETS_DIR}{volume_name}"));
-            }
-
-            if let Some(tls) = &ldap.tls {
-                match &tls.verification {
-                    TlsVerification::Server(TlsServerVerification {
-                        ca_cert: CaCert::SecretClass(cert_secret_class),
-                    }) => {
-                        let volume_name = format!("{authentication_class_name}-tls-certificate");
-
-                        pb.add_volume(build_secret_operator_volume(
-                            &volume_name,
-                            cert_secret_class,
-                            None,
-                        ));
-                        cb.add_volume_mount(&volume_name, format!("{CERTS_DIR}{volume_name}"));
-                    }
-                    // Explicitly listing other possibilities to not oversee new enum variants in the future
-                    TlsVerification::None {}
-                    | TlsVerification::Server(TlsServerVerification {
-                        ca_cert: CaCert::WebPki {},
-                    }) => {}
-                }
-            }
-
+            ldap.add_volumes_and_mounts(pb, vec![cb]);
             Ok(())
         }
         _ => AuthenticationClassProviderNotSupportedSnafu {
             authentication_class_provider: authentication_class.spec.provider.to_string(),
-            authentication_class: ObjectRef::<AuthenticationClass>::new(authentication_class_name),
+            authentication_class: ObjectRef::<AuthenticationClass>::new(
+                authentication_class.metadata.name.as_ref().unwrap(),
```

Comment:
I *think* this should exist, you may need a trait import
```suggestion
                authentication_class.name_unchecked(),
```

Reply:
I changed it :+1: